src/agent/border_agent.cpp

```diff
@@ -227,29 +228,91 @@ static const char *ThreadVersionToString(uint16_t aThreadVersion)
+    const otExtendedPanId *  extPanId    = otThreadGetExtendedPanId(instance);
+    const otExtAddress *     extAddr     = otLinkGetExtendedAddress(instance);
+    const char *             networkName = otThreadGetNetworkName(instance);
+    Mdns::Publisher::TxtList txtList{{"rv", "1"}};
+
+    txtList.emplace_back("nn", networkName);
+    txtList.emplace_back("xp", extPanId->m8, sizeof(extPanId->m8));
+    txtList.emplace_back("tv", ThreadVersionToString(otThreadGetVersion()));
+
+    // "dd" represents for device discriminator which
+    // should always be the IEEE 802.15.4 extended address.
+    txtList.emplace_back("dd", extAddr->m8, sizeof(extAddr->m8));
+
+    state.mConnectionMode = kConnectionModePskc;
+    state.mAvailability   = kAvailabilityHigh;
+#if OTBR_ENABLE_BACKBONE_ROUTER
+    state.mBbrIsActive  = otBackboneRouterGetState(instance) != OT_BACKBONE_ROUTER_STATE_DISABLED;
+    state.mBbrIsPrimary = otBackboneRouterGetState(instance) == OT_BACKBONE_ROUTER_STATE_PRIMARY;
+#endif
+    switch (otThreadGetDeviceRole(instance))
+    {
+    case OT_DEVICE_ROLE_DISABLED:
+        state.mThreadIfStatus = kThreadIfStatusNotInitialized;
+        break;
+    case OT_DEVICE_ROLE_DETACHED:
+        state.mThreadIfStatus = kThreadIfStatusInitialized;
+        break;
+    default:
+        state.mThreadIfStatus = kThreadIfStatusActive;
+    }
+
+    stateUint32 = htobe32(state.ToUint32());
+    txtList.emplace_back("sb", reinterpret_cast<uint8_t *>(&stateUint32), sizeof(stateUint32));
+
+    if (state.mThreadIfStatus == kThreadIfStatusActive)
+    {
+        otError              error;
+        otOperationalDataset activeDataset;
+        uint64_t             activeTimestamp;
+        uint32_t             partitionId;
+
+        if ((error = otDatasetGetActive(instance, &activeDataset)) != OT_ERROR_NONE)
+        {
+            otbrLog(OTBR_LOG_WARNING, "failed to get active dataset: %s", otThreadErrorToString(error));
+        }
+        else
+        {
+            activeTimestamp = htobe64(activeDataset.mActiveTimestamp);
+            txtList.emplace_back("at", reinterpret_cast<uint8_t *>(&activeTimestamp), sizeof(activeTimestamp));
+        }
+
+        partitionId = otThreadGetPartitionId(instance);
+        txtList.emplace_back("pt", reinterpret_cast<uint8_t *>(&partitionId), sizeof(partitionId));
+    }
+
+#if OTBR_ENABLE_BACKBONE_ROUTER
+    {
+        otError                error;
+        otBackboneRouterConfig bbrConfig;
+
+        if ((error = otBackboneRouterGetPrimary(instance, &bbrConfig)) != OT_ERROR_NONE)
+        {
+            otbrLog(OTBR_LOG_WARNING, "failed to get primary BBR state: %s", otThreadErrorToString(error));
+        }
+        else
+        {
+            txtList.emplace_back("sq", &bbrConfig.mSequenceNumber, sizeof(bbrConfig.mSequenceNumber));
+        }
+
+        txtList.emplace_back("dn", otThreadGetDomainName(instance));
+    }
+#endif
+
+    // TODO: append BBR sequence number and BBR port.
```

Comment:
nit: sequence number is appended already
```suggestion
    // TODO: append BBR port.
```

Fyi, BBR port is 61631 by default so far, and not configurable so far.
https://github.com/openthread/openthread/blob/3852ac3d3e10602470a52d98c5c243595c8d9eb2/src/core/backbone_router/backbone_tmf.hpp#L44